### PR TITLE
feat: support init SDK with custom configuration

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
- [x] PR title and description conform to [Pull Request](https://github.com/awslabs/clickstream-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Description of changes:*
1. support init SDK with custom configuration
2. support add global attribute in configuration
3. compatible with amplifyconfiguration.json file

*How did you test these changes?*
added test case for support configure amplify multiple times.

*Documentation update required?*
- [ ] No
- [x] Yes (Please include a PR link for the documentation update)

*General Checklist*
- [x] Added Unit Tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,10 +78,36 @@ public void onCreate() {
     } 
 }
 ```
+#### 3.1 Initialize the SDK with global attributes and custom configuration
```
Comment (zxkane) [inline README.md:81]: should be `3.2`?

File: clickstream/src/test/java/software/aws/solution/clickstream/AnalyticsEventTest.java [hunk 2]
```diff
@@ -51,13 +51,10 @@ public void init() {
      * @return AnalyticsClient
      */
     public static AnalyticsClient getAnalyticsClient() {
-        AWSClickstreamPluginConfiguration.Builder configurationBuilder = AWSClickstreamPluginConfiguration.builder();
-        configurationBuilder.withEndpoint(
-            "http://click-serve-HCJIDWGD3S9F-1166279006.ap-southeast-1.elb.amazonaws.com/collect");
-        AWSClickstreamPluginConfiguration clickstreamPluginConfiguration = configurationBuilder.build();
+        ClickstreamConfiguration configuration = ClickstreamConfiguration.getDefaultConfiguration()
+            .withEndpoint("http://click-serve-HCJIDWGD3S9F-1166279006.ap-southeast-1.elb.amazonaws.com/collect");
```
Comment (zxkane) [inline clickstream/src/test/java/software/aws/solution/clickstream/AnalyticsEventTest.java:55]: replace with `example.com`

**Review by zxkane — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (zhu-xiaowei) [reply README.md]: fixed

File: clickstream/src/test/java/software/aws/solution/clickstream/AnalyticsEventTest.java (hunk 2 above)
Comment (zhu-xiaowei) [reply clickstream/src/test/java/software/aws/solution/clickstream/AnalyticsEventTest.java]: fixed

**Review by zxkane — APPROVED** [review]

**Pull request merged**